modules/phase_field/src/utils/PolycrystalICTools.C

```diff
@@ -251,28 +256,26 @@ PolycrystalICTools::buildElementalGrainAdjacencyGraph(
 
       if (!set_intersection.empty())
       {
-        adjacency_matrix[i][j] = true;
-        adjacency_matrix[j][i] = true;
+        adjacency_matrix(i, j) = true;
```

Comment:
If I'm not mistaken you template the matrix on `Real`, but assign `bool`s here. You might as well template it on `PETSCScalar`, right? I'm not too bothered by the implicit cast, but the whole point of the underlying type seems to be compatible with PETSc.